doc/code/curves.md

```diff
@@ -0,0 +1,217 @@
+# Curves
+
+*Curves* are data structures that manage changes to data over time. More precisely,
+they do not only store data but time-value pairs (keyframes) that allow the retrieval
+of a curve data value for any point in time.
+
+Curves are an integral part of openage's event-based game simulation.
+
+1. [Motivation](#motivation)
+2. [Architecture](#architecture)
+3. [Curve Types](#curve-types)
+   1. [Primitive](#primitive)
+      1. [Common Operations](#common-operations)
+      2. [Discrete](#discrete)
+      3. [Continuous](#continuous)
+      4. [Segmented](#segmented)
+   2. [Container](#container)
+      1. [Queue](#queue)
+      2. [Unordered Map](#unordered-map)
+
+
+## Motivation
+
+The openage curve structures are inspired by a [similar implementation](https://www.forrestthewoods.com/blog/tech_of_planetary_annihilation_chrono_cam/#.lmxbu3vld) from the game Planetary Annihilation.
+
+Curves intend to solve synchronicity problems in games. In traditional implementations
+like lockstep, data is changed incrementally in regular intervals (ticks). Keeping
+these changes and the overall gamestate consistent/in sync over network or across threads
+is very important, as one missing change can result in a desync of the entire simulation.
```

Comment:
but curves can get out of sync the same way, i don't think curves help against desync, but they make playback more smooth and time independent

Reply:
what I mean is that a desync doesn't have drastic, unrecoverable consequences. It can make handling desyncs easier and smoother.